expression/builtin_time_test.go

```diff
@@ -1427,6 +1427,49 @@ func (s *testEvaluatorSuite) TestSecToTime(c *C) {
 	}
 }
 
+func (s *testEvaluatorSuite) TestConvertTz(c *C) {
+	tests := []struct {
+		t       interface{}
+		fromTz  string
+		toTz    string
+		Success bool
+		expect  string
+	}{
+		{"2004-01-01 12:00:00.111", "-00:00", "+12:34", true, "2004-01-02 00:34:00.111"},
+		{"2004-01-01 12:00:00.11", "+00:00", "+12:34", true, "2004-01-02 00:34:00.11"},
+		{"2004-01-01 12:00:00.11111111111", "-00:00", "+12:34", true, "2004-01-02 00:34:00.111111"},
+		{"2004-01-01 12:00:00", "GMT", "MET", true, "2004-01-01 13:00:00"},
+		{"2004-01-01 12:00:00", "-01:00", "-12:00", true, "2004-01-01 01:00:00"},
+		{"2004-01-01 12:00:00", "-00:00", "+13:00", true, "2004-01-02 01:00:00"},
+		{"2004-01-01 12:00:00", "-00:00", "-13:00", true, ""},
+		{"2004-01-01 12:00:00", "-00:00", "-12:88", true, ""},
+		{"2004-01-01 12:00:00", "+10:82", "GMT", false, ""},
+		{"2004-01-01 12:00:00", "+00:00", "GMT", true, ""},
+		{"2004-01-01 12:00:00", "GMT", "+00:00", true, ""},
+		{20040101, "+00:00", "+10:32", true, "2004-01-01 10:32:00"},
+		{3.14159, "+00:00", "+10:32", false, ""},
```

Comment:
@simon-xia 
I'm not sure whether the following 2 unsual cases will keep consistent with MySQL?
mysql> select convert_tz("20040101.111", "+00:00", "+10:32");
+------------------------------------------------+
| convert_tz("20040101.111", "+00:00", "+10:32") |
+------------------------------------------------+
| 2004-01-01 21:33:00                            |
+------------------------------------------------+
1 row in set (0.00 sec)

mysql> select convert_tz(20040101.111, "+00:00", "+10:32");
+----------------------------------------------+
| convert_tz(20040101.111, "+00:00", "+10:32") |
+----------------------------------------------+
| 2004-01-01 10:32:00.000                      |
+----------------------------------------------+
1 row in set, 1 warning (0.00 sec)